morecantile/models.py

```diff
@@ -383,6 +419,21 @@ def sort_tile_matrices(cls, v):
         """Sort matrices by identifier"""
         return sorted(v, key=lambda m: int(m.id))
 
+    @cached_property
+    def is_quadtree(self) -> bool:
+        """Check for quadtree support."""
+        return check_quadkey_support(self.tileMatrices)
```

Comment:
using cached property is recommended by pydantic

Reply:
> Also, this decorator requires that the __dict__ attribute on each instance be a mutable mapping. This means it will not work with some types, such as metaclasses (since the __dict__ attributes on type instances are read-only proxies for the class namespace), and those that specify __slots__ without including __dict__ as one of the defined slots (as such classes don’t provide a __dict__ attribute at all).

Maybe we should check ☝️